pinot-core/src/main/java/org/apache/pinot/core/data/manager/offline/DimensionTableDataManager.java

```diff
@@ -187,43 +221,79 @@ private DimensionTable createFastLookupDimensionTable() {
     Preconditions.checkState(CollectionUtils.isNotEmpty(primaryKeyColumns),
         "Primary key columns must be configured for dimension table: %s", _tableNameWithType);
 
-    Map<PrimaryKey, GenericRow> lookupTable = new HashMap<>();
     List<SegmentDataManager> segmentDataManagers = acquireAllSegments();
     try {
+      // count all documents to limit map re-sizings
```

Comment:
Not introduced in this PR, but the loading logic is specific to the dimension table implementation. Should we consider moving the loading logic into the implementation class?

Reply:
Refactoring whole or a part into static method results in really ugly code (lots of references and passed variables). I'd rather keep it as is.